### PR TITLE
RUBY-3451 Release signing

### PR DESCRIPTION
**Pull request by jamis**
Moves the release process for the Ruby driver to GitHub actions. Previous rake tasks and scripts related to releases have been removed.

The process for creating a new release for the driver is simpler now. Once the `master` branch represents the finished release, manually trigger the "Driver Release" workflow in GitHub. This will:

1. Create a signed tag with the version number in `mongo/version.rb`
2. Build the new gem file
3. Create a detached signature for the gem via GPG
4. Create a draft release for the new version on GitHub
5. Upload the gemfile and the signature to the draft release
6. Push the gem to Rubygems

At that point, all that remains is to manually edit the draft release so the changelog is accurate, and publish the release.

### CODE REVIEW

**Review by alcaeus — COMMENTED** [review]
Workflow LGTM, and an additional 👍 on modifying the rake tasks to provide information if they are invoked locally.

We should also include instructions on how users can verify package integrity. For inspiration, you can look at [PHPC's "Release Integrity"](https://github.com/mongodb/mongo-php-driver?tab=readme-ov-file#release-integrity) chapter in the readme. PHP's extension installer won't verify the signature and will actually download the file itself, so I've included instructions for security-conscious people how to download, verify, and install from our release assets. The process here may be quite similar.

**Comment by jamis** [comment]
Great point, @alcaeus. Further, we should probably hang a lantern on the fact that we're not using Rubygems gem-signing for this, and talk briefly about why that is. I'll update the PR today.

File: README.md [hunk 1]
```diff
@@ -5,7 +5,43 @@ MongoDB Ruby Driver
...
+4. Download the corresponding detached signature file from the [same release](https://github.com/mongodb/mongo-ruby-driver/releases). Look at the bottom of the release that corresponds to the gem file, under the 'Assets' list, for a `.sig` file with the same version number as the gem you wish to install.
+5. Verify the gem with `gpg --verify mongo-X.Y.Z.gem.sig mongo-X.Y.Z.gem` (replacing `X.Y.Z` with the actual version number).
+
+You are looking for text like "Good signature from "MongoDB Ruby Driver Release Signing Key <packaging@mongodb.com>" in the output. If you see that, the signature was found to correspond to the given gem file.
+
+(Note that other output, like "This key is not certified with a trusted signature!", is related to *web of trust* and depends on how strongly you, personally, trust the `ruby-driver.asc` key that you downloaded from us. To learn more, see https://www.gnupg.org/gph/en/manual/x334.html)
+
+### Why not use RubyGems' gem-signing functionality?
```
Comment (alcaeus) [inline README.md:36]: Great explanation! 👏

**Review by alcaeus — APPROVED** [review]
GitHub release workflows LGTM, deferring to @comandeo-mongo for the ruby specifics.

**Review by comandeo-mongo — APPROVED** [review]

**Pull request merged**